staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -137,115 +157,138 @@ func NewCmdAnnotate(parent string, f cmdutil.Factory, ioStreams genericclioption
+		overwrite:       flags.overwrite,
+		resourceVersion: flags.resourceVersion,
+		Recorder:        genericclioptions.NoopRecorder{},
+		selector:        flags.Selector,
+	}
+
 	var err error
 
-	o.RecordFlags.Complete(cmd)
-	o.Recorder, err = o.RecordFlags.ToRecorder()
+	flags.RecordFlags.Complete(cmd)
+	options.Recorder, err = flags.RecordFlags.ToRecorder()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	o.outputFormat = cmdutil.GetFlagString(cmd, "output")
-	o.dryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	options.dryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
 	dynamicClient, err := f.DynamicClient()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.dryRunVerifier = resource.NewQueryParamVerifier(dynamicClient, f.OpenAPIGetter(), resource.QueryParamDryRun)
 
-	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.dryRunStrategy)
-	printer, err := o.PrintFlags.ToPrinter()
+	options.dryRunVerifier = resource.NewQueryParamVerifier(dynamicClient, f.OpenAPIGetter(), resource.QueryParamDryRun)
+
+	cmdutil.PrintFlagsWithDryRunStrategy(flags.PrintFlags, options.dryRunStrategy)
+	printer, err := flags.PrintFlags.ToPrinter()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.PrintObj = func(obj runtime.Object, out io.Writer) error {
+	options.PrintObj = func(obj runtime.Object, out io.Writer) error {
 		return printer.PrintObj(obj, out)
 	}
 
-	if o.list && len(o.outputFormat) > 0 {
-		return fmt.Errorf("--list and --output may not be specified together")
-	}
-
-	o.namespace, o.enforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	options.namespace, options.enforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.builder = f.NewBuilder()
-	o.unstructuredClientForMapping = f.UnstructuredClientForMapping
+	options.builder = f.NewBuilder()
+	options.unstructuredClientForMapping = f.UnstructuredClientForMapping
 
 	// retrieves resource and annotation args from args
 	// also checks args to verify that all resources are specified before annotations
 	resources, annotationArgs, err := cmdutil.GetResourcesAndPairs(args, "annotation")
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.resources = resources
-	o.newAnnotations, o.removeAnnotations, err = parseAnnotations(annotationArgs)
+	options.resources = resources
+	options.newAnnotations, options.removeAnnotations, err = parseAnnotations(annotationArgs)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	return nil
-}
-
-// Validate checks to the AnnotateOptions to see if there is sufficient information run the command.
-func (o AnnotateOptions) Validate() error {
-	if o.all && len(o.selector) > 0 {
-		return fmt.Errorf("cannot set --all and --selector at the same time")
+	// Checks the options and flags to see if there is sufficient information run the command.
+	if flags.List && len(flags.OutputFormat) > 0 {
```

Comment:
Would it be better or possible if we move the rest of this complete function into `validate`?. Because as far as I can see, all the rest is validation.

Reply:
yeah this has been a tricky part. I decided to go this way, as I was looking at other PRs that split flags from options. If you are ok, and the others folks do not object I would to leave it as it is.